flamingpy/noise/cv.py

```diff
@@ -0,0 +1,649 @@
+
+        Process the measurement outcomes of the jth macronode in self.egraph to
+        populate the corresponding reduced node in reduced_graph with a bit value
+        and conditional phase error probability.
+
+        This method modifies self.reduced_graph.
+        """
+        macro_graph = self.egraph
+        delta = self._delta
+        star_index = self.permuted_inds[j]
+        vertex = self.to_points[star_index]
+
+        # Here, j corresponds to the macronode and i to to micronode.
+        # i ranges from 1 to 4, to align with manuscript.
+        verts_and_inds = [self._neighbor_of_micro_i_macro_j(i, j) for i in (1, 2, 3, 4)]
+        neighbors = [tup[0] if tup else None for tup in verts_and_inds]
+        body_indices = [tup[1] if tup else None for tup in verts_and_inds]
+
+        # Array of arrays of measurement outcomes in all the
+        # macronodes adjacent to j.
+        m_arr = np.array([self._hom_outcomes(neighbors[i - 1]) for i in (1, 2, 3, 4)])
+        # Array of processed q-homodyne outcomes from neighboring
+        # macronodes of the form [0, Z(1), Z(2), Z(3), Z(4)].
+        Z_arr = np.array(
+            [0]
+            + [
+                self._process_neighboring_outcomes(m_arr[i - 1], body_indices[i - 1])
+                for i in (1, 2, 3, 4)
+            ]
+        )
+        # p-homodyne outcome of the star node.
+        star_p_val = macro_graph.nodes[vertex]["hom_val_p"]
+
+        # Types of state for the four micronodes directly neighboring
+        # macronode j.
+        types = [
+            macro_graph.nodes[neighbor]["state"] if neighbor else None for neighbor in neighbors
+        ]
+        # Phase error probability and number of p-squeezed states
+        # among the four micronodes in the vicinity of macronode j
+        p_err = 0
+        num_p = 0
+        outcome = 2 * star_p_val
+        gkp_inds = []
+        for i in (1, 2, 3, 4):
+            if types[i - 1] == "p":
+                num_p += 1
+                outcome -= Z_arr[i]
+            if types[i - 1] == "GKP":
+                if delta > 0:
+                    p_err += Z_err_cond(2 * delta, Z_arr[i], use_hom_val=True)
+                gkp_inds += [i]
+        if delta > 0:
+            p_err += Z_err_cond(2 * (2 + num_p) * delta, outcome, use_hom_val=True)
+        p_err = min(p_err, 0.5)
+        p_err = max(p_err, 0)
+
+        bitp = GKP_binner([outcome])[0]
+        bitq = GKP_binner(Z_arr[gkp_inds].astype(np.float64)) if gkp_inds else 0
+
+        processed_bit_val = (bitp + np.sum(bitq)) % 2
+
+        # Update the reduced RHG lattice with the effective
+        # homodyne value and the phase error probability.
+        central_vert = tuple(round(i) for i in vertex)
+        self.reduced_graph.nodes[central_vert]["bit_val"] = processed_bit_val
+        self.reduced_graph.nodes[central_vert]["p_phase_cond"] = p_err
+
+    def reduce(self, noise_model):
```

Comment:
What is the common convention? Ranked by importance, alphabetical, etc.?

Reply:
I prefer by importance and then by alphabetical in case of ties. This is my personal preference. However, in this case, reduce is the function that users will call 99% of the times. It deserves a better spot :smile: